python/pyspark/conf.py

```diff
@@ -203,6 +203,18 @@ def setAll(self, pairs: List[Tuple[str, str]]) -> "SparkConf":
             self.set(k, v)
         return self
 
+    @overload
+    def get(self, key: str) -> Optional[str]:
+        ...
+
+    @overload
+    def get(self, key: str, defaultValue: None) -> Optional[str]:
+        ...
+
+    @overload
+    def get(self, key: str, defaultValue: str) -> str:
+        ...
+
```

Comment:
These are added to clearly indicate which calls can result in `None` and, in turn, avoid ignores or casts in `rdd.py`.

Reply:
qq: Maybe that is a stupid question, but I'd like to ask if
`def get(self, key: str, defaultValue: Optional[str]) -> Optional[str]:`
can't cover the both
`def get(self, key: str, defaultValue: None) -> Optional[str]:`
and
`def get(self, key: str, defaultValue: str) -> str:` ??

Reply:
It covers both, but doesn't capture the same relationship between arguments.  With `(SparkConf, str, str) -> str` we know that

```python
conf.get("foo", "42") 
```

is `str`, which saves as cast / ignores / asserts not None later.

With only `(SparkConf, str, Optional[str]) -> Optional[str]`  we still have to assert that result is not `None`.

(There might be other way of capturing this through type parameters, i.e. `(SparkConf, str, T) -> T` where `T` is `TypeVar("T", None, str)`)